src/main/scala/beam/agentsim/agents/ridehail/allocation/RideHailResourceAllocationManager.scala

```diff
@@ -176,17 +176,18 @@ abstract class RideHailResourceAllocationManager(private val rideHailManager: Ri
           case None =>
             NoVehicleAllocated(request)
         }
-    }.toList
+    }.toVector
     VehicleAllocations(allocResponses)
   }
 
   val repositioningManager: RepositioningManager = createRepositioningManager()
   logger.info(s"Using ${repositioningManager.getClass.getSimpleName} as RepositioningManager")
 
-  def findDepotsForVehiclesInNeedOfRefueling(cavOnly: Boolean = true): Vector[(Id[Vehicle], ParkingStall)] = {
-    val idleVehicleIdsAndLocation: Vector[(Id[Vehicle], RideHailAgentLocation)] =
```

Comment:
This seems dangerous to push back to the caller. If the filtering of excluded is required then it should be handled inside of the implementation itself.

Reply:
@JustinPihony not sure why this is dangerous? I've changed the contract - now you have to provide `idleVehicles`. Maybe I should rename it just to `vehicles`?

Reply:
@colinsheppard what do you think?

Reply:
Yah....if the contract is fine to change then I'm ok with that - but it is *definitely* a contract change - if we are ok with the potential for non-idle and exclusionary coming in in other means then that's fine. But if it should always be idle excluded then it is not ok.